Mathlib/Tactic/FinCases.lean

```diff
@@ -116,8 +123,6 @@ produces three goals with hypotheses
 -/
 
 /- TODO: In mathlib3 we ran `norm_num` when there is no `with` clause. Is this still useful? -/
-/- TODO: can we name the cases generated according to their values,
-   rather than `tail.tail.tail.head`? -/
 
```

Comment:
Could you add a test showing that the cases are named? You could use something like `#guard_msgs in example ... some setup ... fin_cases ... trace_state`.

Reply:
Like this?
```lean
/--
info: case 0.0
⊢ True

case 0.1
⊢ True

case 1.0
⊢ True

case 1.1
⊢ True
-/
#guard_msgs in
example (x y:Fin 2): True := by
  fin_cases x, y
  trace_state
  all_goals simp
```

Reply:
Yes, exactly.

Can you check whether you can select these cases by name too? Like `case 0.0 => ...`? If not, we should use `Name.str` instead of `Name.num`. The `Name.num` is for internal names, so I suspect this case selection won't work.

Reply:
Good point. Changed to `.str`, so that case names now appear as `case «0».«1»` and `case «0».«1» => ...` works